client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import request from 'superagent';
+import defer from 'lodash/defer';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import {
+	LOGIN_TWOFACTOR_UPDATE_NONCE,
+	LOGIN_TWOFACTOR_PUSH_POLL_COMPLETED,
+	LOGIN_TWOFACTOR_PUSH_POLL_START,
+} from 'state/action-types';
+import {
+	getTwoFactorUserId,
+	getTwoFactorAuthNonce,
+	getTwoFactorPushToken,
+	getTwoFactorRememberMe,
+	getTwoFactorPushPollInProgress
+} from 'state/login/selectors';
+
+/***
+ * Module constants
+ */
+const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
+
+/***
+ * Checks the status of the push notification auth
+ *
+ * @param {Object}   store  Global redux store
+ * @returns {Promise}		Promise of result from the API
+ */
+const doAppPushRequest = ( store ) => {
+	return request.post( config( 'two_step_authentication_xhr' ) )
+		.withCredentials()
+		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
+		.accept( 'application/json' )
+		.send( {
+			user_id: getTwoFactorUserId( store.getState() ),
+			two_step_nonce: getTwoFactorAuthNonce( store.getState() ),
+			remember_me: getTwoFactorRememberMe( store.getState() ),
+			two_step_push_token: getTwoFactorPushToken( store.getState() ),
+			client_id: config( 'wpcom_signup_id' ),
+			client_secret: config( 'wpcom_signup_key' ),
+		} ).then( () => {
+			store.dispatch( { type: LOGIN_TWOFACTOR_PUSH_POLL_COMPLETED } );
+		} ).catch( error => {
+			store.dispatch( { type: LOGIN_TWOFACTOR_UPDATE_NONCE, twoStepNonce: error.response.body.data.two_step_nonce } );
+			return Promise.reject( error );
+		} );
+};
+
+/***
+ * Polling the login API for the status of the push notification.
+ * The polling will stop on success or when store's polling progress
+ * state changes to `false`
+ *
+ * @param {Object}   store  Global redux store
+ */
+const doAppPushPolling = store => {
+	let retryCount = 0;
+	const retry = () => {
+		// if polling was stopped or not in progress - stop
+		if ( ! getTwoFactorPushPollInProgress( store.getState() ) ) {
+			return;
+		}
+
+		retryCount++;
+		setTimeout(
```

Comment:
It's stopping because of the condition above:
```js
if ( ! getTwoFactorPushPollInProgress( store.getState() ) ) {
	return;
}
```
Which happens when `LOGIN_TWOFACTOR_PUSH_POLL_STOP` is dispatched